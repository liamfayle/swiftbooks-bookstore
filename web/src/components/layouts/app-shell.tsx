"use client";

import { FC, ReactNode } from "react";
import { AppShell, Burger, Title } from "@mantine/core";
import Image from "next/image";
import { useDisclosure } from "@mantine/hooks";
import cx from "clsx";
import Link from "next/link";

import { CartMenu } from "@/components/cart/cart-menu";
import { AccountMenu } from "@/components/account/account-menu";
import { HeaderLinks } from "@/components/layouts/header-links";
import { Navbar } from "@/components/layouts/navbar";

import logo from "@/images/logo.png";

import classes from "./app-shell.module.css";

export const AppShellLayout: FC<{ children: ReactNode }> = ({ children }) => {
  const [opened, { toggle, close }] = useDisclosure();

  return (
    <AppShell
      header={{ height: 60 }}
      navbar={{ width: 300, breakpoint: "sm", collapsed: { desktop: true, mobile: !opened } }}
    >
      <AppShell.Header className={classes.header}>
        <div className={classes.headerInner}>
          <div className={classes.headerBranding}>
            <Burger opened={opened} onClick={toggle} hiddenFrom="sm" size="sm" />

            <Link href="/">
              <Image
                src={logo}
                alt="Swift Books logo"
                width={40}
                height={40}
                className={classes.desktopLogo}
              />
            </Link>
            <Title order={3} className={cx(classes.title, classes.headerTitle)}>
              Swift <span>Books</span>
            </Title>
          </div>

          <Link href="/">
            <Image
              src={logo}
              alt="Swift Books logo"
              width={40}
              height={40}
              className={classes.mobileLogo}
            />
          </Link>

          <HeaderLinks className={classes.links} />

          <div className={classes.headerActions}>
            <CartMenu className={classes.headerAction} />

            <AccountMenu className={classes.headerAction} />
          </div>
        </div>
      </AppShell.Header>

      <Navbar onClose={close} />

      <AppShell.Main>
        <div className={classes.container}>{children}</div>
      </AppShell.Main>
    </AppShell>
  );
};
