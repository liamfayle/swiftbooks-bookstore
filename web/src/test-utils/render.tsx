import { ReactNode } from "react";
import { render as testingLibraryRender } from "@testing-library/react";
import { MantineProvider } from "@mantine/core";
import { QueryClientProvider } from "@tanstack/react-query";

import { theme } from "@/theme";
import { client } from "@/api/api";
import { AuthProvider, InitGuard } from "@/components/auth/auth-context";

jest.mock("next/navigation", () => ({
  useRouter() {
    return {
      prefetch: () => null,
      replace: () => null,
    };
  },
  usePathname() {
    return "/";
  },
  useSearchParams() {
    return new URLSearchParams();
  },
}));

export function render(ui: ReactNode) {
  return testingLibraryRender(<>{ui}</>, {
    wrapper: ({ children }: { children: ReactNode }) => (
      <MantineProvider theme={theme}>
        <QueryClientProvider client={client}>
          <AuthProvider>
            <InitGuard>{children}</InitGuard>
          </AuthProvider>
        </QueryClientProvider>
      </MantineProvider>
    ),
  });
}
