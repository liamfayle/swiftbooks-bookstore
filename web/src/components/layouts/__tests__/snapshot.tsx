import { render } from "@/test-utils";
import { AppShellLayout } from "../app-shell";

describe("Layouts", () => {
  it("renders app shell layout unchanged", () => {
    const { container } = render(
      <AppShellLayout>
        <span>Content</span>
      </AppShellLayout>,
    );
    expect(container).toMatchSnapshot();
  });
});
