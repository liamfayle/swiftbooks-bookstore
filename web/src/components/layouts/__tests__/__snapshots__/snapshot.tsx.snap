// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts renders app shell layout unchanged 1`] = `
<div>
  <style
    data-mantine-styles="true"
  >
    :root{--mantine-font-family: fontFamily;--mantine-font-family-headings: fontFamily;--mantine-primary-color-filled: var(--mantine-color-teal-filled);--mantine-primary-color-filled-hover: var(--mantine-color-teal-filled-hover);--mantine-primary-color-light: var(--mantine-color-teal-light);--mantine-primary-color-light-hover: var(--mantine-color-teal-light-hover);--mantine-primary-color-light-color: var(--mantine-color-teal-light-color);}:root[data-mantine-color-scheme="dark"]{--mantine-color-anchor: #38d9a9;--mantine-color-dark-filled: #1A1B1E;--mantine-color-dark-filled-hover: #141517;--mantine-color-dark-light: rgba(44, 46, 51, 0.15);--mantine-color-dark-light-hover: rgba(44, 46, 51, 0.2);--mantine-color-dark-light-color: #909296;--mantine-color-dark-outline: #5c5f66;--mantine-color-dark-outline-hover: rgba(92, 95, 102, 0.05);--mantine-color-gray-filled: #495057;--mantine-color-gray-filled-hover: #343a40;--mantine-color-gray-light: rgba(173, 181, 189, 0.15);--mantine-color-gray-light-hover: rgba(173, 181, 189, 0.2);--mantine-color-gray-light-color: #e9ecef;--mantine-color-gray-outline: #dee2e6;--mantine-color-gray-outline-hover: rgba(222, 226, 230, 0.05);--mantine-color-red-filled: #f03e3e;--mantine-color-red-filled-hover: #e03131;--mantine-color-red-light: rgba(255, 107, 107, 0.15);--mantine-color-red-light-hover: rgba(255, 107, 107, 0.2);--mantine-color-red-light-color: #ffc9c9;--mantine-color-red-outline: #ffa8a8;--mantine-color-red-outline-hover: rgba(255, 168, 168, 0.05);--mantine-color-pink-filled: #d6336c;--mantine-color-pink-filled-hover: #c2255c;--mantine-color-pink-light: rgba(240, 101, 149, 0.15);--mantine-color-pink-light-hover: rgba(240, 101, 149, 0.2);--mantine-color-pink-light-color: #fcc2d7;--mantine-color-pink-outline: #faa2c1;--mantine-color-pink-outline-hover: rgba(250, 162, 193, 0.05);--mantine-color-grape-filled: #ae3ec9;--mantine-color-grape-filled-hover: #9c36b5;--mantine-color-grape-light: rgba(204, 93, 232, 0.15);--mantine-color-grape-light-hover: rgba(204, 93, 232, 0.2);--mantine-color-grape-light-color: #eebefa;--mantine-color-grape-outline: #e599f7;--mantine-color-grape-outline-hover: rgba(229, 153, 247, 0.05);--mantine-color-violet-filled: #7048e8;--mantine-color-violet-filled-hover: #6741d9;--mantine-color-violet-light: rgba(132, 94, 247, 0.15);--mantine-color-violet-light-hover: rgba(132, 94, 247, 0.2);--mantine-color-violet-light-color: #d0bfff;--mantine-color-violet-outline: #b197fc;--mantine-color-violet-outline-hover: rgba(177, 151, 252, 0.05);--mantine-color-indigo-filled: #4263eb;--mantine-color-indigo-filled-hover: #3b5bdb;--mantine-color-indigo-light: rgba(92, 124, 250, 0.15);--mantine-color-indigo-light-hover: rgba(92, 124, 250, 0.2);--mantine-color-indigo-light-color: #bac8ff;--mantine-color-indigo-outline: #91a7ff;--mantine-color-indigo-outline-hover: rgba(145, 167, 255, 0.05);--mantine-color-blue-filled: #1c7ed6;--mantine-color-blue-filled-hover: #1971c2;--mantine-color-blue-light: rgba(51, 154, 240, 0.15);--mantine-color-blue-light-hover: rgba(51, 154, 240, 0.2);--mantine-color-blue-light-color: #a5d8ff;--mantine-color-blue-outline: #74c0fc;--mantine-color-blue-outline-hover: rgba(116, 192, 252, 0.05);--mantine-color-cyan-filled: #1098ad;--mantine-color-cyan-filled-hover: #0c8599;--mantine-color-cyan-light: rgba(34, 184, 207, 0.15);--mantine-color-cyan-light-hover: rgba(34, 184, 207, 0.2);--mantine-color-cyan-light-color: #99e9f2;--mantine-color-cyan-outline: #66d9e8;--mantine-color-cyan-outline-hover: rgba(102, 217, 232, 0.05);--mantine-color-teal-filled: #0ca678;--mantine-color-teal-filled-hover: #099268;--mantine-color-teal-light: rgba(32, 201, 151, 0.15);--mantine-color-teal-light-hover: rgba(32, 201, 151, 0.2);--mantine-color-teal-light-color: #96f2d7;--mantine-color-teal-outline: #63e6be;--mantine-color-teal-outline-hover: rgba(99, 230, 190, 0.05);--mantine-color-green-filled: #37b24d;--mantine-color-green-filled-hover: #2f9e44;--mantine-color-green-light: rgba(81, 207, 102, 0.15);--mantine-color-green-light-hover: rgba(81, 207, 102, 0.2);--mantine-color-green-light-color: #b2f2bb;--mantine-color-green-outline: #8ce99a;--mantine-color-green-outline-hover: rgba(140, 233, 154, 0.05);--mantine-color-lime-filled: #74b816;--mantine-color-lime-filled-hover: #66a80f;--mantine-color-lime-light: rgba(148, 216, 45, 0.15);--mantine-color-lime-light-hover: rgba(148, 216, 45, 0.2);--mantine-color-lime-light-color: #d8f5a2;--mantine-color-lime-outline: #c0eb75;--mantine-color-lime-outline-hover: rgba(192, 235, 117, 0.05);--mantine-color-yellow-filled: #f59f00;--mantine-color-yellow-filled-hover: #f08c00;--mantine-color-yellow-light: rgba(252, 196, 25, 0.15);--mantine-color-yellow-light-hover: rgba(252, 196, 25, 0.2);--mantine-color-yellow-light-color: #ffec99;--mantine-color-yellow-outline: #ffe066;--mantine-color-yellow-outline-hover: rgba(255, 224, 102, 0.05);--mantine-color-orange-filled: #f76707;--mantine-color-orange-filled-hover: #e8590c;--mantine-color-orange-light: rgba(255, 146, 43, 0.15);--mantine-color-orange-light-hover: rgba(255, 146, 43, 0.2);--mantine-color-orange-light-color: #ffd8a8;--mantine-color-orange-outline: #ffc078;--mantine-color-orange-outline-hover: rgba(255, 192, 120, 0.05);}:root[data-mantine-color-scheme="light"]{--mantine-color-anchor: #0ca678;--mantine-color-dark-filled: #1A1B1E;--mantine-color-dark-filled-hover: #141517;--mantine-color-dark-light: rgba(26, 27, 30, 0.1);--mantine-color-dark-light-hover: rgba(26, 27, 30, 0.12);--mantine-color-dark-light-color: #1A1B1E;--mantine-color-dark-outline: #1A1B1E;--mantine-color-dark-outline-hover: rgba(26, 27, 30, 0.05);--mantine-color-gray-filled: #495057;--mantine-color-gray-filled-hover: #343a40;--mantine-color-gray-light: rgba(73, 80, 87, 0.1);--mantine-color-gray-light-hover: rgba(73, 80, 87, 0.12);--mantine-color-gray-light-color: #495057;--mantine-color-gray-outline: #495057;--mantine-color-gray-outline-hover: rgba(73, 80, 87, 0.05);--mantine-color-red-filled: #f03e3e;--mantine-color-red-filled-hover: #e03131;--mantine-color-red-light: rgba(240, 62, 62, 0.1);--mantine-color-red-light-hover: rgba(240, 62, 62, 0.12);--mantine-color-red-light-color: #f03e3e;--mantine-color-red-outline: #f03e3e;--mantine-color-red-outline-hover: rgba(240, 62, 62, 0.05);--mantine-color-pink-filled: #d6336c;--mantine-color-pink-filled-hover: #c2255c;--mantine-color-pink-light: rgba(214, 51, 108, 0.1);--mantine-color-pink-light-hover: rgba(214, 51, 108, 0.12);--mantine-color-pink-light-color: #d6336c;--mantine-color-pink-outline: #d6336c;--mantine-color-pink-outline-hover: rgba(214, 51, 108, 0.05);--mantine-color-grape-filled: #ae3ec9;--mantine-color-grape-filled-hover: #9c36b5;--mantine-color-grape-light: rgba(174, 62, 201, 0.1);--mantine-color-grape-light-hover: rgba(174, 62, 201, 0.12);--mantine-color-grape-light-color: #ae3ec9;--mantine-color-grape-outline: #ae3ec9;--mantine-color-grape-outline-hover: rgba(174, 62, 201, 0.05);--mantine-color-violet-filled: #7048e8;--mantine-color-violet-filled-hover: #6741d9;--mantine-color-violet-light: rgba(112, 72, 232, 0.1);--mantine-color-violet-light-hover: rgba(112, 72, 232, 0.12);--mantine-color-violet-light-color: #7048e8;--mantine-color-violet-outline: #7048e8;--mantine-color-violet-outline-hover: rgba(112, 72, 232, 0.05);--mantine-color-indigo-filled: #4263eb;--mantine-color-indigo-filled-hover: #3b5bdb;--mantine-color-indigo-light: rgba(66, 99, 235, 0.1);--mantine-color-indigo-light-hover: rgba(66, 99, 235, 0.12);--mantine-color-indigo-light-color: #4263eb;--mantine-color-indigo-outline: #4263eb;--mantine-color-indigo-outline-hover: rgba(66, 99, 235, 0.05);--mantine-color-blue-filled: #1c7ed6;--mantine-color-blue-filled-hover: #1971c2;--mantine-color-blue-light: rgba(28, 126, 214, 0.1);--mantine-color-blue-light-hover: rgba(28, 126, 214, 0.12);--mantine-color-blue-light-color: #1c7ed6;--mantine-color-blue-outline: #1c7ed6;--mantine-color-blue-outline-hover: rgba(28, 126, 214, 0.05);--mantine-color-cyan-filled: #1098ad;--mantine-color-cyan-filled-hover: #0c8599;--mantine-color-cyan-light: rgba(16, 152, 173, 0.1);--mantine-color-cyan-light-hover: rgba(16, 152, 173, 0.12);--mantine-color-cyan-light-color: #1098ad;--mantine-color-cyan-outline: #1098ad;--mantine-color-cyan-outline-hover: rgba(16, 152, 173, 0.05);--mantine-color-teal-filled: #0ca678;--mantine-color-teal-filled-hover: #099268;--mantine-color-teal-light: rgba(12, 166, 120, 0.1);--mantine-color-teal-light-hover: rgba(12, 166, 120, 0.12);--mantine-color-teal-light-color: #0ca678;--mantine-color-teal-outline: #0ca678;--mantine-color-teal-outline-hover: rgba(12, 166, 120, 0.05);--mantine-color-green-filled: #37b24d;--mantine-color-green-filled-hover: #2f9e44;--mantine-color-green-light: rgba(55, 178, 77, 0.1);--mantine-color-green-light-hover: rgba(55, 178, 77, 0.12);--mantine-color-green-light-color: #37b24d;--mantine-color-green-outline: #37b24d;--mantine-color-green-outline-hover: rgba(55, 178, 77, 0.05);--mantine-color-lime-filled: #74b816;--mantine-color-lime-filled-hover: #66a80f;--mantine-color-lime-light: rgba(116, 184, 22, 0.1);--mantine-color-lime-light-hover: rgba(116, 184, 22, 0.12);--mantine-color-lime-light-color: #74b816;--mantine-color-lime-outline: #74b816;--mantine-color-lime-outline-hover: rgba(116, 184, 22, 0.05);--mantine-color-yellow-filled: #f59f00;--mantine-color-yellow-filled-hover: #f08c00;--mantine-color-yellow-light: rgba(245, 159, 0, 0.1);--mantine-color-yellow-light-hover: rgba(245, 159, 0, 0.12);--mantine-color-yellow-light-color: #f59f00;--mantine-color-yellow-outline: #f59f00;--mantine-color-yellow-outline-hover: rgba(245, 159, 0, 0.05);--mantine-color-orange-filled: #f76707;--mantine-color-orange-filled-hover: #e8590c;--mantine-color-orange-light: rgba(247, 103, 7, 0.1);--mantine-color-orange-light-hover: rgba(247, 103, 7, 0.12);--mantine-color-orange-light-color: #f76707;--mantine-color-orange-outline: #f76707;--mantine-color-orange-outline-hover: rgba(247, 103, 7, 0.05);}
  </style>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <style
    data-mantine-styles="inline"
  >
    :root{--app-shell-navbar-width:calc(18.75rem * var(--mantine-scale));--app-shell-navbar-offset:calc(18.75rem * var(--mantine-scale));--app-shell-header-height:calc(3.75rem * var(--mantine-scale));--app-shell-header-offset:calc(3.75rem * var(--mantine-scale));--app-shell-padding:0px;}@media(min-width: 48em){:root{--app-shell-navbar-transform:translateX(calc(var(--app-shell-navbar-width) * -1));--app-shell-navbar-transform-rtl:translateX(var(--app-shell-navbar-width));--app-shell-navbar-offset:0px !important;}}@media(max-width: 47.99375em){:root{--app-shell-navbar-width:100%;--app-shell-navbar-offset:0px;--app-shell-navbar-transform:translateX(calc(var(--app-shell-navbar-width) * -1));--app-shell-navbar-transform-rtl:translateX(var(--app-shell-navbar-width));}}
  </style>
  <div
    class="m-89ab340 mantine-AppShell-root"
    data-resizing="true"
    style="--app-shell-transition-duration: 200ms; --app-shell-transition-timing-function: ease;"
  >
    <header
      class="m-3b16f56b mantine-AppShell-header right-scroll-bar-position header"
      data-with-border="true"
      style="--app-shell-header-z-index: 100;"
    >
      <div
        class="headerInner"
      >
        <div
          class="headerBranding"
        >
          <button
            class="mantine-focus-auto m-fea6bf1a mantine-Burger-root m-87cf2631 mantine-UnstyledButton-root mantine-hidden-from-sm"
            data-size="sm"
            style="--burger-size: var(--burger-size-sm);"
            type="button"
          >
            <div
              class="m-d4fb9cad mantine-Burger-burger"
              data-reduce-motion="true"
            />
          </button>
          <a
            href="/"
          >
            <img
              alt="Swift Books logo"
              class="desktopLogo"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </a>
          <h3
            class="title headerTitle m-8a5d1357 mantine-Title-root"
            data-order="3"
            style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size);"
          >
            Swift 
            <span>
              Books
            </span>
          </h3>
        </div>
        <a
          href="/"
        >
          <img
            alt="Swift Books logo"
            class="mobileLogo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </a>
        <div
          class="linkContainer links"
        >
          <a
            class="mantine-focus-auto m-849cf0da link m-b6d8b162 mantine-Text-root mantine-Anchor-root"
            data-underline="hover"
            href="/books"
          >
            Books
          </a>
          <div
            class="separator"
          />
          <a
            class="mantine-focus-auto m-849cf0da link m-b6d8b162 mantine-Text-root mantine-Anchor-root"
            data-underline="hover"
            href="/book-lists"
          >
            Book lists
          </a>
        </div>
        <div
          class="headerActions"
        >
          <div
            class="indicator m-e5262200 mantine-Indicator-root"
            data-variant="light"
            style="--indicator-color: var(--mantine-color-violet-filled); --indicator-size: calc(1rem * var(--mantine-scale)); --indicator-top: calc(0.1875rem * var(--mantine-scale)); --indicator-right: calc(0.1875rem * var(--mantine-scale)); --indicator-translate-x: 50%; --indicator-translate-y: -50%;"
          >
            <button
              class="mantine-focus-auto mantine-active headerAction m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
              data-variant="subtle"
              style="--ai-size: calc(2.5rem * var(--mantine-scale)); --ai-bg: transparent; --ai-hover: rgba(0, 0, 0, 0.12); --ai-color: black; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
              type="button"
            >
              <svg
                class="tabler-icon tabler-icon-shopping-bag"
                fill="none"
                height="2.5rem"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="2.5rem"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M6.331 8h11.339a2 2 0 0 1 1.977 2.304l-1.255 8.152a3 3 0 0 1 -2.966 2.544h-6.852a3 3 0 0 1 -2.965 -2.544l-1.255 -8.152a2 2 0 0 1 1.977 -2.304z"
                />
                <path
                  d="M9 11v-5a3 3 0 0 1 6 0v5"
                />
              </svg>
            </button>
          </div>
          <button
            aria-controls="account-popover-dropdown"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="mantine-focus-auto mantine-active headerAction m-8d3f4000 mantine-ActionIcon-root m-87cf2631 mantine-UnstyledButton-root"
            data-variant="subtle"
            id="account-popover-target"
            style="--ai-size: calc(2.5rem * var(--mantine-scale)); --ai-bg: transparent; --ai-hover: rgba(0, 0, 0, 0.12); --ai-color: black; --ai-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
            type="button"
          >
            <svg
              class="tabler-icon tabler-icon-user"
              fill="none"
              height="2.5rem"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="2.5rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 7a4 4 0 1 0 8 0a4 4 0 0 0 -8 0"
              />
              <path
                d="M6 21v-2a4 4 0 0 1 4 -4h4a4 4 0 0 1 4 4v2"
              />
            </svg>
          </button>
        </div>
      </div>
    </header>
    <nav
      class="m-45252eee mantine-AppShell-navbar container mantine-hidden-from-sm"
      data-with-border="true"
      style="--app-shell-navbar-z-index: calc(100 + 1);"
    >
      <h3
        class="title m-8a5d1357 mantine-Title-root"
        data-order="3"
        style="--title-fw: var(--mantine-h3-font-weight); --title-lh: var(--mantine-h3-line-height); --title-fz: var(--mantine-h3-font-size);"
      >
        Swift 
        <span>
          Books
        </span>
      </h3>
      <div
        class="items"
      >
        <hr
          class="hr"
        />
        <a
          class="mantine-focus-auto m-849cf0da link link m-b6d8b162 mantine-Text-root mantine-Anchor-root"
          data-underline="hover"
          href="/books"
        >
          <svg
            class="tabler-icon tabler-icon-book-2"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19 4v16h-12a2 2 0 0 1 -2 -2v-12a2 2 0 0 1 2 -2h12z"
            />
            <path
              d="M19 16h-12a2 2 0 0 0 -2 2"
            />
            <path
              d="M9 8h6"
            />
          </svg>
           
          Books
        </a>
        <a
          class="mantine-focus-auto m-849cf0da link link m-b6d8b162 mantine-Text-root mantine-Anchor-root"
          data-underline="hover"
          href="/book-lists"
        >
          <svg
            class="tabler-icon tabler-icon-books"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 4m0 1a1 1 0 0 1 1 -1h2a1 1 0 0 1 1 1v14a1 1 0 0 1 -1 1h-2a1 1 0 0 1 -1 -1z"
            />
            <path
              d="M9 4m0 1a1 1 0 0 1 1 -1h2a1 1 0 0 1 1 1v14a1 1 0 0 1 -1 1h-2a1 1 0 0 1 -1 -1z"
            />
            <path
              d="M5 8h4"
            />
            <path
              d="M9 16h4"
            />
            <path
              d="M13.803 4.56l2.184 -.53c.562 -.135 1.133 .19 1.282 .732l3.695 13.418a1.02 1.02 0 0 1 -.634 1.219l-.133 .041l-2.184 .53c-.562 .135 -1.133 -.19 -1.282 -.732l-3.695 -13.418a1.02 1.02 0 0 1 .634 -1.219l.133 -.041z"
            />
            <path
              d="M14 9l4 -1"
            />
            <path
              d="M16 16l3.923 -.98"
            />
          </svg>
           
          Book lists
        </a>
      </div>
    </nav>
    <main
      class="m-8983817 mantine-AppShell-main"
    >
      <div
        class="container"
      >
        <span>
          Content
        </span>
      </div>
    </main>
  </div>
</div>
`;
